js/apps/admin-ui/src/realm-settings/OpenIdEndpointConigurationTab.tsx

```diff
@@ -0,0 +1,175 @@
+  TextInput,
+} from "@patternfly/react-core";
+import { useEffect, useState } from "react";
+import { Controller, useForm } from "react-hook-form";
+import { useTranslation } from "react-i18next";
+import { FormAccess } from "../components/form/FormAccess";
+import { HelpItem } from "ui-shared";
+import { addTrailingSlash } from "../util";
+import { adminClient } from "../admin-client";
+import { MinusIcon, PlusIcon } from "@patternfly/react-icons";
+import { getAuthorizationHeaders } from "../utils/getAuthorizationHeaders";
+
+type RealmSettingsThemesTabProps = {
+  realm: RealmRepresentation;
+  save: (realm: RealmRepresentation) => void;
+};
+
+export const OpenIdEndpointConigurationTab = ({
+  realm,
+  save,
+}: RealmSettingsThemesTabProps) => {
+  const { t } = useTranslation();
+
+  const [newClaim, setNewClaim] = useState<string>("");
+  const { control, handleSubmit, setValue } = useForm<RealmRepresentation>();
+
+  const setupForm = () => {
+    let attributes = realm.attributes || {};
+    
+    if (!("claimsSupported" in attributes)) {
+      getOpenIdEndpointConfiguration().then((result) => {
+        attributes.claimsSupported = result.claims_supported.join(",");
+        setValue("attributes", attributes);
+      });
+    }
+    else{
+      setValue("attributes", attributes);
+    }
+    
+  };
+
+  useEffect(setupForm, []);
+
+  async function getOpenIdEndpointConfiguration(): Promise<any> {
+    const response = await fetch(
+      `${addTrailingSlash(adminClient.baseUrl)}/realms/${
+        realm.realm
+      }/.well-known/openid-configuration`,
+      {
+        method: "GET",
+        headers: getAuthorizationHeaders(await adminClient.getAccessToken()),
+      },
+    );
+
+    if (!response.ok) {
+      throw new Error(
+        `Server responded with invalid status: ${response.statusText}`,
+      );
+    }
+    return response.json();
+  }
+
+  return (
+    <PageSection variant="light">
+      <FormAccess
+        isHorizontal
+        role="manage-realm"
+        className="pf-u-mt-lg"
+        onSubmit={handleSubmit(save)}
+      >
+        <FormGroup
+          label={t("claimsSupported")}
+          fieldId="kc-claims-supported"
+          labelIcon={
+            <HelpItem
+              helpText={t("claimsSupportedHelp")}
+              fieldLabelId="claimsSupported"
+            />
+          }
+        >
+          <Controller
+            name="attributes"
+            control={control}
+            render={({ field }) => {
+              const claimsSupported = field.value?.claimsSupported?.trim()
+                ? field.value.claimsSupported.split(",")
+                : [];
+              return (
+                <>
+                  {claimsSupported.map((claim: string, index: number) => {
+                    if (claim) {
+                      return (
+                        <InputGroup key={index}>
+                          <TextInput
+                            id={"textInput" + index}
```

Comment:
If this `id` is not used it can be removed. If it's needed for the tests we prefer using a test id.
```suggestion
                            data-testid={"textInput" + index}
```

Reply:
Removed all id's as they were not used.